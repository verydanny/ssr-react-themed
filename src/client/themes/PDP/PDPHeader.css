@value ( RentBlue ) from 'ui/shared/themes/colors/colors.css';

.PDPHeader {
  height: 45px;
  background-color: RentBlue;
  display: flex;
}

.PDPHeader_Logo {
  flex-grow: 1;
  text-align: center;
  padding-right: 30px;
}

.PDPHeader_Logo, .PDPHeader_Back {
  margin-top: 15px;
  margin-bottom: 15px;
}

body .PDPHeader_Logo svg {
  height: 17px;
  line-height: 0;
  position: relative;
  right: 10px;
  width: 91px;
}

.PDPHeader_Back {
  margin-left: 19px;
  padding: 0;
  border: 0;
  background-color: RentBlue;
}

body .PDPHeader_Back svg {
  position: relative;
  bottom: 2px;
  height: 16px;
  padding-right: 20px;
  width: 21px;
}
