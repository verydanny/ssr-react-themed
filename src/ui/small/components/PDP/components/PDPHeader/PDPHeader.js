import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { themed } from 'react-themed-too'
import { Icon } from 'ui/shared/components/Icon'
import arrowLeft from 'ui/shared/components/Icon/svgs/arrowLeft.svg'
// TODO: whats wrong with this logo?
// import { pdpRentLogo } from 'ui/shared/components/Icon/svgs/pdp'
import rentComLogo from 'ui/shared/components/Icon/svgs/rentComLogo.svg'
import { buildSearchUrl } from 'lib/url'

class PDPHeader extends PureComponent {
  handleClickBack = () => {
    const {
      locationSlug,
      propertyType,
      pushState,
      refinementSlug,
      searchType,
      viewType,
    } = this.props

    const newPath = buildSearchUrl({
      locationSlug,
      propertyType,
      searchType,
      refinementSlug,
      viewType,
    })

    pushState(newPath)
  }

  render() {
    const {
      theme,
    } = this.props
    return (
      <div
        className={theme.PDPHeader}
        data-tid="pdpHeader"
      >
        <button
          className={theme.PDPHeader_Back}
          onClick={this.handleClickBack}
          data-tid="pdpHeader_Back"
        >
          <Icon svgs={{ arrowLeft }} />
        </button>
        <a
          className={theme.PDPHeader_Logo}
          href="/"
          data-tid="pdpHeader_Logo"
        >
          <Icon svgs={{ rentComLogo }} />
        </a>
      </div>
    )
  }
}

PDPHeader.propTypes = {
  locationSlug: PropTypes.string,
  propertyType: PropTypes.string,
  pushState: PropTypes.func,
  refinementSlug: PropTypes.string,
  searchType: PropTypes.string,
  theme: PropTypes.object,
  viewType: PropTypes.string,
}

PDPHeader.defaultProps = {
  theme: {},
}

export default themed(/^PDPHeader/, { pure: true })(PDPHeader)
