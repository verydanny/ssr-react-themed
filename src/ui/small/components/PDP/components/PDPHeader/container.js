import { connect } from 'react-redux'
import pushState from 'app/store/thunks/pushState'
import {
  locationSlug as getLocationSlug,
  propertyType as getPropertyType,
  getSearchType,
  getViewType,
} from 'app/selectors/criteria'
import {
  refinementSlug,
} from 'app/selectors/refinements'
import PDPHeader from './PDPHeader'

const mapStateToProps = state => ({
  locationSlug: getLocationSlug(state),
  propertyType: getPropertyType(state),
  refinementSlug: refinementSlug(state),
  searchType: getSearchType(state),
  viewType: getViewType(state),
  // queryParams,
  // shouldClearBBox,
  // shouldClearPage,
})

const mapDispatchToProps = {
  pushState,
}

export default connect(mapStateToProps, mapDispatchToProps)(PDPHeader)
