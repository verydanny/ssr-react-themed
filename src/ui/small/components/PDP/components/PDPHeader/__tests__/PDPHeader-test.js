import React from 'react'
import { mount } from 'enzyme'
import PDPHeader from '../PDPHeader'

describe('Click on adjust filters', () => {
  const props = {
    theme: {},
    pushState() {},
  }

  it('renders header with rentlogo and back button', () => {
    const wrapper = mount(
      <PDPHeader {...props} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
