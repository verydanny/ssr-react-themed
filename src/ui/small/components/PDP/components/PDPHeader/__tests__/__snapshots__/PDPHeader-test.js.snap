// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click on adjust filters renders header with rentlogo and back button 1`] = `
<Themed(PDPHeader)
  pushState={[Function]}
  theme={Object {}}
>
  <PDPHeader
    pushState={[Function]}
    theme={Object {}}
  >
    <div
      data-tid="pdpHeader"
    >
      <button
        data-tid="pdpHeader_Back"
        onClick={[Function]}
      >
        <Themed(Icon)
          svgs={
            Object {
              "arrowLeft": "",
            }
          }
        >
          <Icon
            svgs={
              Object {
                "arrowLeft": "",
              }
            }
            theme={Object {}}
          />
        </Themed(Icon)>
      </button>
      <a
        data-tid="pdpHeader_Logo"
        href="/"
      >
        <Themed(Icon)
          svgs={
            Object {
              "rentComLogo": "",
            }
          }
        >
          <Icon
            svgs={
              Object {
                "rentComLogo": "",
              }
            }
            theme={Object {}}
          />
        </Themed(Icon)>
      </a>
    </div>
  </PDPHeader>
</Themed(PDPHeader)>
`;
